### DISCUSSION TITLE
Is there any way to setQueryData() while deleting elements from the existing array stored in the store?

### DISCUSSION THREAD

**Discussion by sonjaerock** [TanStack/query#4458] [Q&A]
### Describe the bug
Removed todo rendered via setQueryData() in onMutate() after calling delete api in sample todolist. It remains deleted in the store, but it is still displayed.
Is there any way to setQueryData() while deleting elements from the existing array stored in the store?

### Your minimal, reproducible example
https://codesandbox.io/s/long-tdd-iur84y?file=/src/Components/TodoList.tsx

### Steps to reproduce

1. useQuery(Array)
2. setQueryData(removedArray)
3. no-rerender

### Expected behavior

rerender

### Platform

OS: windows
Browser: chrome

### react-query version

v3.39.2

### TypeScript version

v4.8.4

### Additional context

test

**Comment by TkDodo** [comment]
I don't understand the question and the reproduction is far from minimal. Can you elaborate please?

**Comment by Zerebokep** [comment]
This is most likely because you didn't update the data immutably.
